vllm/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -72,6 +70,8 @@ def __init__(self,
         if not custom_ar:
             # disable because of missing custom allreduce library
             # e.g. in a non-cuda environment
+            logger.warning("Custom allreduce is disabled because "
```

Comment:
This is a good addition.

Reply:
I think this should only be a warning on CUDA and RoCM platforms. IIUC this will always log a warning on other platforms and this should not be a warning e.g. on TPUs.

Also, could you change the comment on line 72 to `# e.g. in a non-GPU environment`

Reply:
Ok, I guess this is only constructed in the `cuda_communicator`, so not concerned about TPUs now.

However, it does look like this will always log a warning on AMD GPUs earlier than MI300X, and we shouldn't be warning users about expected behavior.

Reply:
The custom allreduce library can be built on AMD GPU earlier than MI300X. We just don't enable it. So there shouldn't be a warning as we will not try to create `CustomAllreduce` object.